### ISSUE TITLE
Cannot install build-essential, gcc, and ruby

### ISSUE THREAD

**Issue by adeeb1** [Microsoft/BashOnWindows#664]
Description:

I am trying to set up a new Ruby on Rails environment, and I cannot install Ruby 2.1.8 using RVM. I also cannot install other packages, such as `build-essential` and the `gcc` compiler. I was able to install these packages in previous Windows Insider builds.

I don't know if it caused issues, but before I upgraded from build 14385, I uninstalled Bash on Ubuntu on Windows and then reinstalled it after the upgrade.

Expected results:

I should be able to install these packages using aptitude on Ubuntu.

Results:

```
~$ sudo apt-get install build-essential

Reading package lists... Done
Building dependency tree
Reading state information... Done
The following extra packages will be installed:
  binutils cpp cpp-4.8 dpkg-dev fakeroot g++ g++-4.8 gcc gcc-4.8
  libalgorithm-diff-perl libalgorithm-diff-xs-perl libalgorithm-merge-perl
  libasan0 libatomic1 libc-dev-bin libc6-dev libdpkg-perl libfakeroot
  libfile-fcntllock-perl libgcc-4.8-dev libgomp1 libitm1 libquadmath0
  libstdc++-4.8-dev libtsan0 linux-libc-dev make manpages-dev
Suggested packages:
  binutils-doc cpp-doc gcc-4.8-locales debian-keyring g++-multilib
  g++-4.8-multilib gcc-4.8-doc libstdc++6-4.8-dbg gcc-multilib automake1.9
  libtool flex bison gdb gcc-doc gcc-4.8-multilib libgcc1-dbg libgomp1-dbg
  libitm1-dbg libatomic1-dbg libasan0-dbg libtsan0-dbg libquadmath0-dbg
  glibc-doc libstdc++-4.8-doc make-doc
The following NEW packages will be installed:
  binutils build-essential cpp cpp-4.8 dpkg-dev fakeroot g++ g++-4.8 gcc
  gcc-4.8 libalgorithm-diff-perl libalgorithm-diff-xs-perl
  libalgorithm-merge-perl libasan0 libatomic1 libc-dev-bin libc6-dev
  libdpkg-perl libfakeroot libfile-fcntllock-perl libgcc-4.8-dev libgomp1
  libitm1 libquadmath0 libstdc++-4.8-dev libtsan0 linux-libc-dev make
  manpages-dev
0 upgraded, 29 newly installed, 0 to remove and 37 not upgraded.
1 not fully installed or removed.
Need to get 0 B/38.8 MB of archives.
After this operation, 112 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
dpkg: error processing package gcc-4.8-base:amd64 (--configure):
 package gcc-4.8-base:amd64 is not ready for configuration
 cannot configure (current status `half-installed')
Errors were encountered while processing:
 gcc-4.8-base:amd64
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Windows build number: 14390

Steps / commands required to reproduce the error:

`sudo apt-get install build-essential`

**Comment by eldadh** [comment]
try sudo apt-get install -f

**Comment by adeeb1** [comment]
@eldadh I did, but unfortunately that didn't work.

**Comment by benhillis** [comment]
Interesting, I have not seen this previously.  Do you know what you did before you got into this state?

I did some searching and found this thread on askubuntu, could you give this a try?
http://askubuntu.com/questions/490671/fix-half-installed-package

**Comment by adeeb1** [comment]
I was getting my Rails environment set up, but something about my whole setup on the terminal was not right, so when I came back to finish it later, I saw that build 14390 was available. I uninstalled Bash on Windows via the CMD Prompt since I wanted to get a fresh start, and then I upgraded to build 14390. Then I reinstalled Bash on Windows using the "Turn on or off Windows features" screen. After that, I installed RVM as usual, and I ran into the issue above when attempting to install Ruby.
